test/index.js

```diff
@@ -1921,3 +1947,66 @@ test('apiMiddleware must dispatch a failure FSA on an unsuccessful API call with
   t.plan(8);
   actionHandler(anAction);
 });
+
+test('apiMiddleware must use a [RSAA].fetch custom fetch wrapper when present', t => {
+  const asyncWorker = async () => 'Done!';
+  const responseBody = {
+    id: 1,
+    name: 'Alan',
+    error: false
+  };
+  const api = nock('http://127.0.0.1')
+    .get('/api/users/1')
+    .reply(200, responseBody);
+  const anAction = {
+    [RSAA]: {
+      endpoint: 'http://127.0.0.1/api/users/1',
+      method: 'GET',
+      fetch: async (endpoint, opts) => {
+        t.pass('custom fetch handler called');
+
+        // Simulate some async process like retrieving cache
+        await asyncWorker();
+
+        const res = await fetch(endpoint, opts);
+        const json = await res.json();
+
+        return new Response(
+          JSON.stringify({
+            ...json,
+            foo: 'bar'
+          }),
+          {
+            // Example of custom `res.ok`
```

Comment:
#170 is addressing the ability to customize "ok" behavior per-request, what's going on here?

Reply:
`res.ok` is true when the [status is 200-299](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok), so this is just an example of setting that boolean with something aside from the actual response status code.

Reply:
Gotcha. The issue I linked to is proposing a way to pass in your own 'ok' check instead of `res.ok` so I was just confused seeing this here. Carry on!